src/main/java/uk/ac/cam/cl/dtg/segue/api/AdminFacade.java

```diff
@@ -281,6 +294,112 @@ public synchronized Response modifyUsersRole(@Context final HttpServletRequest r
     return Response.ok().build();
   }
 
+  /**
+   * This method will allow users to have their teacher pending status mass changed.
+   *
+   * @param request - to help determine access rights.
+   * @param status    - new teacher pending status.
+   * @param userIds - a list of user ids to change en-mass
+   * @return Success shown by returning an ok response
+   */
+  @POST
+  @Path("/users/teacher_pending/{status}")
+  @Produces(MediaType.APPLICATION_JSON)
+  @Consumes(MediaType.APPLICATION_JSON)
+  public synchronized Response modifyUsersTeacherPendingStatus(@Context final HttpServletRequest request,
+                                                               @PathParam("status") final Boolean status,
+                                                               final List<Long> userIds) {
+
+    try {
+      RegisteredUserDTO requestingUser = userManager.getCurrentRegisteredUser(request);
+      if (!isUserAnAdminOrEventManager(userManager, requestingUser)) {
+        return new SegueErrorResponse(Status.FORBIDDEN, ACCESS_DENIED_MESSAGE)
+            .toResponse();
+      }
+
+      Map<String, List<Long>> failedUpdates = new HashMap<>();
+
+      for (Long userId : userIds) {
+        modifyTeacherPendingStatusForUser(userId, requestingUser, status, failedUpdates);
+      }
+
+      if (!failedUpdates.isEmpty()) {
+        String errorMessage = "";
+
+        if (failedUpdates.containsKey(USERS_NOT_FOUND)) {
+          errorMessage = String.format("One or more users could not be found: %s. ",
+              failedUpdates.get(USERS_NOT_FOUND));
+        }
+
+        if (failedUpdates.containsKey(FAILED_TO_SEND)) {
+          errorMessage += String.format("Emails could not be sent to userIds: %s",
+              failedUpdates.get(FAILED_TO_SEND));
+        }
+        return new SegueErrorResponse(Status.BAD_REQUEST, errorMessage).toResponse();
```

Comment:
A user not found could be considered a bad request, yes, but the other option (failed to send email) sounds like a server error, not a user error.

Reply:
Essentially what I was trying to achieve was that it wouldn't cause the front end to change into a server error page if an email failed to send so I was trying to avoid a 500 error (also only wanted it to give an error after having gone through the for loop of users, not erroring out part way through). If the error is under 500 the user (in this case an ADMIN user) gets a popup on the front end to inform them of an error and any message details.